### PR TITLE
SED-2145 auto-number-of-threads-is-consumed-even-when-only-one-thread…

### PR DESCRIPTION
**Pull request by david-stephan**
…-is-required

### CODE REVIEW

File: step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/ForBlockHandler.java [hunk 1]
```diff
@@ -152,7 +153,7 @@ public Consumer<DataPoolRow> createWorkItemConsumer(WorkerController<DataPoolRow
 						}
 					};
 				}
-			}, numberOfThreads);
+			}, numberOfThreads, OptionalInt.empty());
```
Comment (jeromecomte) [inline step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/ForBlockHandler.java:156]: Would recommend to add a new method consumeWork with the additional parameter in order to keep the signature and avoid this change here

File: step-plans/step-plans-core/src/main/java/step/threadpool/ThreadPool.java [hunk 2]
```diff
@@ -135,7 +138,8 @@ public boolean isParallel() {
 	}
 
 	public <WORK_ITEM> void consumeWork(Iterator<WORK_ITEM> workItemIterator,
-			WorkerItemConsumerFactory<WORK_ITEM> workItemConsumerFactory, int numberOfThreads) {
+			WorkerItemConsumerFactory<WORK_ITEM> workItemConsumerFactory, int numberOfThreads,
+										OptionalInt expectedNumberOfThreads) {
```
Comment (jeromecomte) [inline step-plans/step-plans-core/src/main/java/step/threadpool/ThreadPool.java:142]: Would rename the parameter as follow:
- numberOfThreads => specifiedNumberOfThreads
- expectedNumberOfThreads => requiredNumberOfThreads

**Review by jeromecomte — APPROVED** [review]
Approving merge. See minor comments

File: step-plans/step-plans-core/src/main/java/step/threadpool/ThreadPool.java (hunk 2 above)
Comment (david-stephan) [reply step-plans/step-plans-core/src/main/java/step/threadpool/ThreadPool.java]: done

**Pull request merged**